### ISSUE TITLE
Cassiopeia: No skip links available in frontend templates.

### ISSUE THREAD

**Issue by mbeganyi-a11y** [joomla/joomla-cms#48421]
### Steps to reproduce the issue

**NOTE**: Accessibility backlog issue and work-in-progress. This issue has been opened to account for all the issues flagged in the initial [Milestone 1a Audit Report](https://accessibility-project.joomla.org/results/Milestone-1a/milestone1/deliverable1/wcag-report). If this is a duplicate issue, it will be closed by the end of September 2026. 

Milestone 1a   finding ID | M1a-040
-- | --
Backlog   priority | Medium
Severity —   carried from Milestone 1a | Moderate
Implementation   effort | Low
Target   milestone(s) | MS2a, MS3a,   MS3b, MS7b, MS8a, MS10
WCAG 2.2   success criterion | 2.4.1 Bypass   Blocks
Responsibility   tag(s) from report | Development
Affected   Joomla area/component | https://accessibility-project.joomla.org/results/Milestone-1a/milestone1/deliverable1/wcag-report#efc77b1a-b5b9-41e7-b8b9-6ad83f62fbb0
Affected user   journey(s) | 02 – Frontend   Template (Cassiopeia); 10 – Frontend User Forms
Source   report/reference | Milestone 1a   WCAG audit report
STF reporting   reference | To be   assigned
Issue owner | To be   assigned
Status | Triage

<p class=MsoNormal>Audit finding<o:p></o:p></p>

<p class=MsoNormal>No skip links available<o:p></o:p></p>

<p class=MsoNormal>User impact<o:p></o:p></p>

<p class=MsoNormal>The issue creates an accessibility barrier for keyboard
users as they cannot quickly skip repetitive blocks of content.<o:p></o:p></p>

<p class=MsoNormal>Remediation plan<o:p></o:p></p>

<p class=MsoNormal>Provide a skip link mechanism and ensure it is included
consistently in shared templates and layouts.<o:p></o:p></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'>Steps to reproduce</b><o:p></o:p></p>

<p class=MsoNormal>1. Open a Cassiopeia page or frontend form.<o:p></o:p></p>

<p class=MsoNormal>2. Press Tab from the top of the page.<o:p></o:p></p>

<p class=MsoNormal>3. Check whether a skip-to-main-content link appears.<o:p></o:p></p>

<p class=MsoNormal>4. Observe that no usable skip link is available.<o:p></o:p></p>

</body>

</html>

### Version

6.1

### Expected result

Allow keyboard-only users to at least skip the main navigation with a “skip to content” link. Including a skip link to other important parts of the page, for example to the login form, could also be helpful.

### Actual result

In the front-end Cassopeia template, there are no skip links implemented by default. While screen reader and other assistive technology users can navigate by landmarks (banner/navigation/main), this is not possible for keyboard-only users.

### What device were you using? (eg, phone, laptop)

Desktop web

### What web browser were you using? (eg, Chrome, Safari)

Chrome, Firefox, Edge, Safari

### Assistive Technology?

N/A

**Comment by brianteeman** [comment]
This is not true. Just enable the skipto plugin. It does a LOT more than just skipping to main

It is not enabled by default because it was thought it might interfere with a custom template

<img width="1619" height="664" alt="Image" src="https://github.com/user-attachments/assets/4eec740f-08c1-4436-9c07-d51b7aac5b91" />

<img width="1920" height="971" alt="Image" src="https://github.com/user-attachments/assets/8130e478-7da1-4fcf-9aba-38c7df153421" />

**Comment by mbeganyi-a11y** [comment]
I suppose that's why it was flagged in the accessibility audit as it's not there by default. How will new users be made aware of the fact this plugin exists? Is it worth notifying users of the option to use this skipto plugin as part of a required tour of Joomla or something like that? But I'm not sure that's easier than implementing a basic skip link in the templates. What do you think?

**Comment by brianteeman** [comment]
There is a big difference between making joomla an accessible solution for building web sites and forcing fake accessibility settings for the site user. Hard coding a Skipto main link doesn't make your site accessible as it will all depend on how the site is built and structured.all it will do is pass an automated test. I would classify this as something the site builder should be doing. It's no different to alt description for images. We give you the tools to add them - we don't force them just to pass a test. We'd be no different to an accessibility overlay that claims it's accessible because it passes a test.

**Comment by mbeganyi-a11y** [comment]
Couldn't agree more, and I wouldn't suggest mimicking overlays. As I explore the systems further, I see that in the backend, extensive keyboard navigation is enabled by default at the top of the page. If the author hasn't configured skip links, then they won't show in the frontend, which is where this issue is flagged. Given this structure, I don't see a true accessibility violation here, as Joomla as the authoring tool is providing the skip-link mechanisms for website owners to implement. If that functionality wasn't possible, then I could see the issue being valid but in the backend only. Any thoughts?

**Comment by rdeutz** [comment]
I would say can be closed as fixed. For later work we need to keep that in mind e.g. accessible frontend template and accessible sample data.

**Comment by mbeganyi-a11y** [comment]
Closed as per previous comment.
